### PR TITLE
[BPK-3692] Remove NSDate methods in BPKCalendar

### PR DESCRIPTION
**Pull request by k0nserv**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by k0nserv** [comment]
The idea here is to keep the `DONT_USE` methods for the RN calendar implementation. 

I'm also considering changing some of the methods in the delegate to make it better in Swift since we are makign a breaking change anyway

File: Backpack/Calendar/Classes/BPKCalendar.m [hunk 1]
```diff
@@ -39,8 +39,38 @@
 #import "BPKCalendarStickyHeader.h"
 #import "BPKCalendarYearPill.h"
 
+
 NS_ASSUME_NONNULL_BEGIN
 
+// This is kept around only to support the React Native calendar implementation
```
Comment (georgegillams) [inline Backpack/Calendar/Classes/BPKCalendar.m:45]: If we're removing the deprecated method, why not remove this for RN too and we can adopt the breaking change in `backpack-react-native`?

Comment (k0nserv) [reply Backpack/Calendar/Classes/BPKCalendar.m:45]: Actually the current RN implementation actually relies on this API with no other option to migrate to imo

**Review by georgegillams — APPROVED** [review]
LGTM

**Pull request merged**